server/service/async/async.go

```diff
@@ -105,7 +106,7 @@ func storePurgeActiveHostID(pool fleet.RedisPool, zsetKey string, hid uint, repo
     return redis.call('ZREMRANGEBYSCORE', KEYS[1], '-inf', ARGV[3])
   `)
 
-	conn := pool.Get()
+	conn := pool.Get().(apmredigo.Conn).WithContext(ctx)
```

Comment:
I don't think I can do that since the type returned by Wrap() is of Conn, not redis.Conn. I'm honestly expecting people to be against wrapping this all the time since its the same thing as having a middleware, so I doubt we would be fine with passing apmredigo.Conn everywhere instead of redis.Conn. Also this completely prevents using opentelemetry alongside later on if we want.

https://pkg.go.dev/go.elastic.co/apm/module/apmredigo#Wrap

Reply:
This wouldn't work actually. But I have no way of knowing here if tracing is enabled or not or what kind since there is no config object in this section of code.